File: Python/8. String to Integer (atoi).py
class Solution:
    def myAtoi(self, str):
        """
        :type str: str
        :rtype: int
        """
        str = str.strip()
        ans, flag = 0, 1
        for i in range(len(str)):
            if str[i] == '-':
                flag = -1
                i += 1
            elif str[i] == '+':
                i += 1
            while i<len(str) and str[i] >= '0' and str[i] <='9':
                ans = ans*10 + ord(str[i]) - ord('0')
                i += 1
                if ans*flag >= 2147483647:
                    return 2147483647
                if ans*flag <= -2147483648:
                    return -2147483648
            return ans*flag
        return ans

File: C++/8. String to Integer (atoi).cpp
class Solution {
public:
    int myAtoi(string str) {
        int flag = 1;
        long ans = 0;
        for(int i=0;i<str.length();i++)
        {
            if(str[i]==' ')
                continue;
            if(str[i]=='+' || str[i]=='-')
            {
                flag = str[i] == '-' ? -1 : 1;
                i++;
            }
            while(str[i] >= '0' && str[i] <= '9')
            {
                ans = ans*10 + str[i]-'0';
                if(ans*flag >= INT_MAX)
                    return INT_MAX;
                if(ans*flag <= INT_MIN)
                    return INT_MIN;
                i++;
            }
            return ans*flag;
        }
            
        return ans;
    }
};